mozilla_repo_urls/result.py

```diff
@@ -35,4 +35,10 @@ def taskcluster_role_prefix(self) -> str:
         return f"repo:{self.host}/{self.repo_path}"
 
     def __eq__(self, other):
-        return self.normalized == other.normalized
+        if isinstance(other, RepoUrlParsed):
+            return self.normalized == other.normalized
+
+        if isinstance(other, str):
+            return self.normalized == other
```

Comment:
Do we really need this, or can we get away with returning False for mismatching types?